packages/sdk/src/react/ui/modals/BuyModal/index.tsx

```diff
@@ -0,0 +1,105 @@
+import type { Hex } from "viem";
+import { buyModal$ } from "./_store";
+import type { Order } from "../../../_internal";
+import { observer, useSelector } from "@legendapp/state/react";
+import { useCollectible, useCollection } from "../../../hooks";
+import { ActionModal } from "../_internal/components/actionModal";
+import { useEffect } from "react";
+import QuantityInput from "../_internal/components/quantityInput";
+import { useBuyCollectable } from "../../../hooks/useBuyCollectable";
+
+export type ShowBuyModalArgs = {
+  chainId: string;
+  collectionAddress: Hex;
+  tokenId: string;
+  order: Order;
+};
+
+export const useBuyModal = () => {
+  return {
+    show: (args: ShowBuyModalArgs) => buyModal$.open(args),
+    close: () => buyModal$.close(),
+  };
+};
+
+export const BuyModal = () => {
```

Comment:
After closing the modal, same Buy CTA doesn't work to open it again, while clicking on another buy CTA opens. Btw, I'm not sure if we should use animation for this modal, if we will, we should add to others too. 
<video src="https://github.com/user-attachments/assets/ad898e7b-73a5-4f7e-bbf4-d0f67746ed8b" />